internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -180,6 +180,46 @@ func dataSourceKubernetesCluster() *pluginsdk.Resource {
 								},
 							},
 						},
+						"azure_keyvault_secrets_provider": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"enabled": {
+										Type:     pluginsdk.TypeBool,
+										Computed: true,
+									},
+									"secret_rotation_enabled": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"secret_rotation_interval": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"secret_provider_identity": {
```

Comment:
Given the other addons will have their identity blocks renamed to `secret_identity` at a later stage, and the schema of them is identical, we should be consistent here and rename this to `secret_identity` as well.

Reply:
I'm not sure I understand - `secret_identity` sounds like it would be specific for the secret provider addon. Wouldn't the future change involve renaming all addon blocks to something more generic, like `addon_identity`?

Reply:
So apologies, there was a misunderstanding on my part. The identity blocks will be `{thing}_identity` and ideally `thing` is a succinct term for the add-on (so the identity for `ingress_application_gateway` would be renamed to `ingress_identity`). From the block name its clear that it's a provider so `secret_identity` should be sufficient in this case?

Reply:
Done :)